```console
#23 69.66 # github.com/docker/docker/builder/builder-next
#23 69.66 builder/builder-next/builder.go:639:5: unknown field err in struct literal of type errInvalidFilterValue
#23 69.66 builder/builder-next/builder.go:645:5: unknown field err in struct literal of type errInvalidFilterValue
```

:eyes: (related to @corhere's suggestion, I think?)

--- a/builder/builder-next/builder.go
+++ b/builder/builder-next/builder.go
@@ -54,6 +54,12 @@ func (e errConflictFilter) Error() string {
 
 func (errConflictFilter) InvalidParameter() {}
 
+type errInvalidFilterValue struct {
+	error
+}
+
+func (errInvalidFilterValue) InvalidParameter() {}
+
 var cacheFields = map[string]bool{
 	"id":          true,
 	"parent":      true,
@@ -629,11 +635,15 @@ func toBuildkitPruneInfo(opts types.BuildCachePruneOptions) (client.PruneInfo, e
 	case 1:
 		ts, err := timetypes.GetTimestamp(untilValues[0], time.Now())
 		if err != nil {
-			return client.PruneInfo{}, errors.Wrapf(err, "%q filter expects a duration (e.g., '24h') or a timestamp", filterKey)
+			return client.PruneInfo{}, errInvalidFilterValue{
+				errors.Wrapf(err, "%q filter expects a duration (e.g., '24h') or a timestamp", filterKey),
+			}
 		}
 		seconds, nanoseconds, err := timetypes.ParseTimestamps(ts, 0)
 		if err != nil {
-			return client.PruneInfo{}, errors.Wrapf(err, "failed to parse timestamp %s", ts)
+			return client.PruneInfo{}, errInvalidFilterValue{
+				errors.Wrapf(err, "failed to parse timestamp %q", ts),
+			}
 		}
 
 		until = time.Since(time.Unix(seconds, nanoseconds))

--- a/daemon/prune.go
+++ b/daemon/prune.go
@@ -2,7 +2,6 @@ package daemon // import "github.com/docker/docker/daemon"
 
 import (
 	"context"
-	"fmt"
 	"regexp"
 	"strconv"
 	"sync/atomic"
@@ -230,15 +229,15 @@ func getUntilFromPruneFilters(pruneFilters filters.Args) (time.Time, error) {
 	}
 	untilFilters := pruneFilters.Get("until")
 	if len(untilFilters) > 1 {
-		return until, fmt.Errorf("more than one until filter specified")
+		return until, errdefs.InvalidParameter(errors.New("more than one until filter specified"))
 	}
 	ts, err := timetypes.GetTimestamp(untilFilters[0], time.Now())
 	if err != nil {
-		return until, err
+		return until, errdefs.InvalidParameter(err)
 	}
 	seconds, nanoseconds, err := timetypes.ParseTimestamps(ts, 0)
 	if err != nil {
-		return until, err
+		return until, errdefs.InvalidParameter(err)
 	}
 	until = time.Unix(seconds, nanoseconds)
 	return until, nil

